common/app/views/support/HtmlCleaner.scala

```diff
@@ -811,13 +811,17 @@ case class AffiliateLinksCleaner(pageUrl: String, sectionId: String, showAffilia
 }
 
 object AffiliateLinksCleaner {
-  def replaceLinksInHtml(html: Document, pageUrl: String, appendDisclaimer: Boolean, contentType: String, skimlinksId: String): Document = {
-    val links = html.getElementsByAttribute("href")
 
-    val supportedLinks: mutable.Seq[Element] = links.asScala.filter(isAffiliatable)
+  def getAffiliateableLinks(html:Document): mutable.Seq[Element] =
+    html.getElementsByAttribute("href").asScala.filter(isAffiliatable)
+
+  def replaceLinksInHtml(html: Document, pageUrl: String, appendDisclaimer: Option[Boolean], contentType: String, skimlinksId: String): Document = {
+
+    val supportedLinks: mutable.Seq[Element] = getAffiliateableLinks(html)
```

Comment:
does this need to be a mutable Seq?

Reply:
Yep because we mutate it in the line below.